tools/wptserve/wptserve/request.py

```diff
@@ -355,12 +358,12 @@ def __init__(self, request_handler):
         super(H2Request, self).__init__(request_handler)
 
 
-def maybedecode(s):
+def maybe_encode(s):
     if isinstance(s, text_type):
-        return s
+        return s.encode("utf-8")
```

Comment:
Presumably iso-8859-1 in Python is equivalent to isomorphic decode and not to windows-1252. In that case we should just leave it at that and not try to output bytes, I think.

Reply:
I'm not so sure. If I print out the decoded string, the non-Unicode characters are all garbled. Although it could well be the print function does something else based on the charset. I don't really know much about the Python encoding internals.

More fundamentally, the type is wrong; even if it is an isomorphic conversion, the fact that the value now has text type would confuse the users.

Reply:
I don't see why. That's the exact same thing JavaScript does for its networking APIs.

Reply:
Yes, but JavaScript historically only had a single type for byte strings and Unicode strings, and that isn't the case with Python. I don't think using Unicode strings with garbage is the right option.

Reply:
Are they really that much easier to operate on? I mean, almost everything works as it would on an ASCII-only Unicode string?

Reply:
If they are, why would Python 3 encode them? (To me it seems like added complexity over the current situation, having to prefix stuff with b everywhere, whereas now it's all reasonably straightforward.)

Reply:
Strings are in fact easier to work with, mostly because of the Python 2/3 incompatibility. When you try to split, join, concat strings, the other operands (usually string literals) must be the same type as the first. Yet in many places, lots of string literals aren't annotated with either `b` or `u`, and they hence become textual strings in Python 3.